src/Client/Web/Bit.Client.Web.BlazorUI/Components/DatePicker/BitDatePicker.razor.cs

```diff
@@ -156,6 +156,9 @@ protected override void RegisterComponentClasses()
         {
             ClassBuilder.Register(() => IsEnabled is false
                 ? $"{RootElementClass}-disabled-{VisualClassRegistrar()}" : string.Empty);
+
+            ClassBuilder.Register(() => Culture.TextInfo.IsRightToLeft
```

Comment:
since the `Culture` parameter has direct effect on the `ClassBuilder`, it should reset the `ClassBuilder` after setting new values.